src/main/java/uk/gov/register/presentation/resource/NewPagination.java

```diff
@@ -14,28 +16,38 @@ public NewPagination(Optional<Integer> optionalStart, Optional<Integer> optional
         this.totalEntries = totalEntries;
     }
 
-    @SuppressWarnings("unused, used by html")
-    public int getTotalEntries() {
+    @Override
+    public long getTotalEntries() {
         return totalEntries;
     }
 
+    @Override
     public boolean hasNextPage() {
         return start + limit - 1 < totalEntries;
     }
 
+    @Override
     public boolean hasPreviousPage() {
         return start > 1;
     }
 
+    @Override
     public String getPreviousPageLink() {
         return String.format("?start=%s&limit=%s", start - limit, limit);
     }
 
+    @Override
+    public long pageSize() {
+        throw new NotImplementedException("not implemented");
+    }
+
```

Comment:
Can we remove pageSize() from IPagination so that we don't need to override it and throw a NotImplementedException?